src/de/map.rs

```diff
@@ -163,11 +163,11 @@ impl<'de, 'a> de::Deserializer<'de> for MapKey<'a, 'de> {
         self.de.deserialize_str(visitor)
     }
 
-    fn deserialize_string<V>(self, _visitor: V) -> Result<V::Value, Self::Error>
+    fn deserialize_string<V>(self, visitor: V) -> Result<V::Value, Self::Error>
     where
         V: Visitor<'de>,
     {
-        unreachable!()
+        self.de.deserialize_string(visitor)
```

Comment:
Interesting just this is supported, so it will work with:

`{  "foo": "bar" }`
but not
`{ "foo": 42 }`
?

I'd love tests of a few structs from JSON -> parse to map -> to JSON showing the same output (okay, modulo sorting the keys)

Reply:
And tests to show this works when the map values are string, int, option, struct, and array.